src/Entity/WebauthnCredentialRepository.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Entity;
+
+use Doctrine\Persistence\ManagerRegistry;
+use Webauthn\Bundle\Repository\PublicKeyCredentialSourceRepository as BasePublicKeyCredentialSourceRepository;
+use Webauthn\PublicKeyCredentialSource as BasePublicKeyCredentialSource;
+
+final class WebauthnCredentialRepository extends BasePublicKeyCredentialSourceRepository
+{
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, WebauthnCredential::class);
+    }
+
+    public function saveCredentialSource(BasePublicKeyCredentialSource $publicKeyCredentialSource, bool $flush = true): void
+    {
+        if (!$publicKeyCredentialSource instanceof WebauthnCredential) {
+            $publicKeyCredentialSource = new WebauthnCredential(
+                $publicKeyCredentialSource->getPublicKeyCredentialId(),
+                $publicKeyCredentialSource->getType(),
+                $publicKeyCredentialSource->getTransports(),
+                $publicKeyCredentialSource->getAttestationType(),
+                $publicKeyCredentialSource->getTrustPath(),
+                $publicKeyCredentialSource->getAaguid(),
+                $publicKeyCredentialSource->getCredentialPublicKey(),
+                $publicKeyCredentialSource->getUserHandle(),
+                $publicKeyCredentialSource->getCounter()
+            );
+        }
+        parent::saveCredentialSource($publicKeyCredentialSource);
```

Comment:
shouldn't this forward the `$flush` parameter ? Otherwise, it is useless

Reply:
The [parent](https://github.com/web-auth/webauthn-framework/blob/4.6.x/src/symfony/src/Repository/DoctrineCredentialSourceRepository.php#L35) already calls the flush method

Reply:
Ah, the parent does not have this argument at all. So why adding it in the child class without using it ?